torch/__init__.py

```diff
@@ -416,6 +416,9 @@ def use_deterministic_algorithms(mode, *, warn_only=False):
         * :class:`torch.nn.AdaptiveMaxPool2d` when attempting to differentiate a CUDA tensor
         * :class:`torch.nn.FractionalMaxPool2d` when attempting to differentiate a CUDA tensor
         * :class:`torch.nn.FractionalMaxPool3d` when attempting to differentiate a CUDA tensor
+        * :class:`torch.nn.MaxUnpool1d`
```

Comment:
@mruberry, in response to your comment here: https://github.com/pytorch/pytorch/pull/78838#discussion_r971959851

> Shouldn't the warning also be on the functional variants? I think these modules just call their functional counterparts

You're right, the modules do just call their functional counterparts, and that is actually where the alerts are being generated from (I'm almost sure that that's true of all of the modules listed here, not just MaxUnpool). I should add tests for all of the functional variants and add them all to the documentation as well.